config/database.yml

```
@@ -50,5 +50,6 @@ test:
 production:
   <<: *default
   database: medicine_notebook_production
-  username: medicine_notebook
-  password: <%= ENV['MEDICINE_NOTEBOOK_DATABASE_PASSWORD'] %>
+  username: <%= ENV['DB_USERNAME'] %>
+  password: <%= ENV['DB_PASSWORD'] %>
+  socket: socket: <%= ENV['DB_SOCKET_PRO'] %>
```

1つのサーバーでdevelopment、productionの両方を使うことはないのでKEYはDB_SOCKETで良さそうです。
あとsocket:というのが二重になってしまっています。
```suggestion
  socket:  <%= ENV['DB_SOCKET'] %>
```